File: src/253MeetingRoomsII/253MeetingRoomsII.cpp
#include <algorithm>
#include <map>
#include <vector>
using namespace std;

/**
 * Definition for an interval.
 */
struct Interval {
  int start;
  int end;
  Interval() : start(0), end(0) {}
  Interval(int s, int e) : start(s), end(e) {}
};

class Solution {
public:
  int minMeetingRooms(vector<Interval> &intervals) {
    map<int, int> m;
    for (auto &p : intervals) {
      ++m[p.start];
      --m[p.end];
    }
    int global = 0, local = 0;
    for (auto &p : m) {
      local += p.second;
      global = max(global, local);
    }
    return global;
  }
};

int main() { return 0; }

File: src/253MeetingRoomsII/253MeetingRoomsII.rs
use std::collections::BTreeMap;
impl Solution {
    pub fn min_meeting_rooms(intervals: Vec<Vec<i32>>) -> i32 {
        let mut map:BTreeMap<i32, i32> = BTreeMap::new();
        for interval in &intervals {
            if let Some(x) = map.get_mut(&interval[0]){
                *x += 1;
            }else{
                map.insert(interval[0], 1);
            }
            if let Some(x) = map.get_mut(&interval[1]){
                *x -= 1;
            }else{
                map.insert(interval[1], -1);
            }
        }
        let (mut global, mut local) = (0, 0);
        for (key, val) in map.iter() {
            local += val;
            global = std::cmp::max(global, local);
        }
        
        global as i32
    }
}
